### DISCUSSION TITLE
Serving text files from /public or other methods of accessing env vars?

### DISCUSSION THREAD

**Discussion by OskarHulter** [vercel/next.js#22304] [Help]
According to the documentation several file types are supported inside public:
txt, html are called out explicitly. 
One would hope that json is also included but from what I have read that might not be supported at this moment.
(https://github.com/vercel/next.js/issues/8251)

When I try adding files to public I don't see this change reflected in the static dir inside of build folder.
![image](https://user-images.githubusercontent.com/20733377/108380686-5fa63e00-7207-11eb-96fa-6f9317625c64.png)
(Several files added of different types. Only env.json has a real reference - could others be cleaned by tree shaking?)

![image](https://user-images.githubusercontent.com/20733377/108381114-c88db600-7207-11eb-95ce-415afd303aaf.png)
build/static folder content, this does contain the fonts (media) but no other files besides what next generates.

We only need to push a static text-file with env vars so that we can overwrite them using our standard build scripts.

I'm also considering just adding .env's to the repo but our builds require that they can be changed by the build agents at each env. Basically we just need a simple textfile that rides along in the build folder. 

The reason for this is that our architects want to send the same bundle through all envs - not rebuild for each step in the pipeline. This is why we need one text-file inside of the built folder to overwrite the env vars. In normal react projects we have served a env.json inside of public.

I would be really grateful for any suggestions or recommendations on this issue. You probably have some great ideas I haven't considered and I would love to hear your thoughts.

**Comment by michrome** [comment]
I think there's a few separate points here:

1. `.json` files can be _served_ from the `/public` directory (json is text).
2. The issue you're referencing relates to reading files in your project from API routes. There is a workaround which is to use `require`; see [this answer](https://github.com/vercel/next.js/discussions/21666#discussioncomment-322999).
3. Do you want to [build your application](https://nextjs.org/docs/api-reference/cli#build) in different environments (by which I mean CI, test, prod)? You can do that and this is a [useful discussion](https://github.com/vercel/next.js/discussions/22143).
